pkg/apis/scheduling/v1alpha1/types_placement.go

```diff
@@ -0,0 +1,128 @@
+//
+// placement is in Pending state initially, when a location is selected by the placement, the placement
+// turns to UnBound state. In Pending or UnBound state, the selection rule can be updated to select another location.
+// When the user annotates a namespace with the key of "scheudling.kcp.dev/placement", the namespace will
+// pick one placement, and this placement is transfered to Bound state. The spec of the placement is immutable
+// in Bound state. The placement will turns back to Unbound state when no namespace uses this
+// placement any more.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories=kcp
+type Placement struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec PlacementSpec `json:"spec,omitempty"`
+
+	// +optional
+	Status PlacementStatus `json:"status,omitempty"`
+}
+
+type PlacementSpec struct {
+	// loacationSelectors represents a slice of label selector to select a location, these label selectors
+	// are logically ORed.
+	LoacationSelectors []metav1.LabelSelector `json:"locationSelectors,omitempty"`
+
+	// locationResource is the group-version-resource of the instances that are subject to the locations to select.
+	//
+	// +required
+	// +kubebuilder:Required
+	LocationResource GroupVersionResource `json:"locationResource"`
+
+	// namespaceSelector is a label selector to select ns. It match all ns by default, but can be specified to
+	// a certain set of ns.
+	NamespaceSelector metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+
+	// locationWorkspace is an absolute reference to a workspace for the location. If it is not set, the workspace of
+	// APIBinding will be used.
+	// +optional
+	// +kubebuilder:validation:Pattern:="^root(:[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
+	LocationWorkspace *string `json:"locationWorkspace,omitempty"`
+}
+
+type PlacementStatus struct {
+	// phase is the current phase of the placement
+	//
+	// +required
+	// +kubebuilder:Required
+	// +kubebuilder:validation:Enum="";Pending;Bound;UnBound
+	Phase PlacementPhase `json:"phase"`
+
+	// selectedLocation is the location that a picked by this placement.
+	SelectedLocation *LocationReference `json:"selectedLocation,omitempty"`
+
+	// Current processing state of the Placement.
+	// +optional
+	Conditions conditionsv1alpha1.Conditions `json:"conditions,omitempty"`
+}
+
+// LocationReference describes a loaction that are provided in the specified Workspace.
+type LocationReference struct {
+	// path is an absolute reference to a workspace, e.g. root:org:ws. The workspace must
+	// be some ancestor or a child of some ancestor.
+	// +optional
+	// +kubebuilder:validation:Pattern:="^root(:[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
+	Path string `json:"path,omitempty"`
```

Comment:
is this actually required? Same for APIBinding of course. Or do we default it to be the local workspace? Worth to check in code. Defaulting seems to be nice UX.

Reply:
Went ahead and implemented it https://github.com/kcp-dev/kcp/pull/1262.